packages/core/src/actions/transactions/waitForTransaction.ts

```diff
@@ -1,8 +1,20 @@
-import type { providers } from 'ethers'
+import type { TransactionResponse } from '@ethersproject/providers'
+import type { BigNumber, providers } from 'ethers'
 import { toUtf8String } from 'ethers/lib/utils.js'
 
 import type { Hash } from '../../types'
+import { fetchBlockNumber } from '../network-status'
 import { getProvider } from '../providers'
+import { fetchTransaction } from './fetchTransaction'
+
+type EthersReplaceable = {
+  data: string
+  from: string
+  nonce: number
+  to: string
+  value: BigNumber
+  startBlock: number
```

Comment:
Alphabetizing. Would also be good to use `ResolvedConfig['BigNumberType']` from `abitype` instead of `BigNumber` directly for `value`.

```suggestion
  data: string
  from: string
  nonce: number
  startBlock: number
  to: string
  value: BigNumber
```